stan/math/opencl/kernels/matrix_multiply.hpp

```diff
@@ -141,6 +141,125 @@ const local_range_kernel<cl::Buffer, cl::Buffer, cl::Buffer, int, int, int,
     matrix_multiply("matrix_multiply", matrix_multiply_kernel_code,
                     {{"THREAD_BLOCK_SIZE", 32}, {"WORK_PER_THREAD", 8}});
 
+// \cond
+static const char* matrix_vector_multiply_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * Matrix-vector multiplication R=A*B on the OpenCL device
+     *
+     * @param[in] A matrix in matrix-vector multiplication
+     * @param[in] B vector in matrix-vector multiplication
+     * @param[out] R the output vector
+     * @param[in] M Number of rows for matrix A
+     * @param[in] N Number of cols for matrix A and number of rows for vector B
+     * @param[in] lower_upper_A the triangularity of A (lower, upper or none)
+     * @param[in] lower_upper_B the triangularity of B (lower, upper or none)
+     */
+    __kernel void matrix_vector_multiply(
+        const __global double* A, const __global double* B, __global double* R,
+        const int M, const int N, unsigned int lower_upper_A,
+        unsigned int lower_upper_B) {
+      const int gid = get_global_id(0);
+
+      int start = 0;
```

Comment:
Mind changing the start to be `const` like this:
`const int start = lower_upper_A == UPPER ? gid + 1 : 0; `